src/components/AttachmentCarousel.js

```diff
@@ -0,0 +1,231 @@
+import React from 'react';
+import {View} from 'react-native';
+import PropTypes from 'prop-types';
+import {withOnyx} from 'react-native-onyx';
+import _ from 'lodash';
+import Button from './Button';
+import * as Expensicons from './Icon/Expensicons';
+import styles from '../styles/styles';
+import AttachmentView from './AttachmentView';
+import SwipeableView from './SwipeableView';
+import addEncryptedAuthTokenToURL from '../libs/addEncryptedAuthTokenToURL';
+import themeColors from '../styles/themes/default';
+import reportActionPropTypes from '../pages/home/report/reportActionPropTypes';
+import canUseTouchScreen from '../libs/canUseTouchscreen';
+import CONFIG from '../CONFIG';
+import CONST from '../CONST';
+import ONYXKEYS from '../ONYXKEYS';
+
+const propTypes = {
+    /** sourceUrl is used to determine the starting index in the array of attachments */
+    sourceURL: PropTypes.string,
+
+    /** Object of report actions for this report */
+    reportActions: PropTypes.objectOf(PropTypes.shape(reportActionPropTypes)),
+
+    /** Callback to update the parent modal's state with a sourceUrl and name from the attachments array */
+    onArrowPress: PropTypes.func,
+};
+
+const defaultProps = {
+    sourceURL: '',
+    reportActions: {},
+    onArrowPress: () => {},
+};
+
+class AttachmentCarousel extends React.Component {
+    constructor(props) {
+        super(props);
+
+        this.canUseTouchScreen = canUseTouchScreen();
+        this.makeAndSetArrowState = this.makeAndSetArrowState.bind(this);
+        this.cycleThroughAttachments = this.cycleThroughAttachments.bind(this);
+        this.handleArrowPress = this.handleArrowPress.bind(this);
+        this.onShowArrow = this.onShowArrow.bind(this);
+
+        this.state = {showArrows: this.canUseTouchScreen};
+    }
+
+    componentDidMount() {
+        this.makeAndSetArrowState();
+        if (this.canUseTouchScreen) {
+            return;
+        }
+        document.addEventListener('keydown', this.handleArrowPress);
+    }
+
+    componentDidUpdate(prevProps) {
+        const prevReportSize = _.size(prevProps.reportActions);
+        const currReportSize = _.size(this.props.reportActions);
+        if (prevReportSize === currReportSize) {
+            return;
+        }
+        this.makeAndSetArrowState();
+    }
+
+    componentWillUnmount() {
+        if (this.canUseTouchScreen) {
+            return;
+        }
+        document.removeEventListener('keydown', this.handleArrowPress);
+    }
```

Comment:
We should rewrite this component in a way we split native & web code separately as interactions are different based on platforms. Let's not mix all code in a single file and make it clumsy.

Let's keep the Base Component with all functionalities and common codes

Interactions/event listeners should be handled in platform-specific files.

@chiragsalian what do you think?

Reply:
I think the question is, does this component intend to be modified further down the line for other features? Will it be used in another places outside of AttachmentModal? 
Because there's not much code differentiating the mobile vs web event handlers and I feel like it would introduce some redundant code and complexity to separate the two imo. 

As for adding pressable to SwipeableView  in the web component I feel it's a necessary start to introducing web based events with that view. I wouldn't understand why there wouldn't be some drag and drop functionality to mirror the UX in mobile eventually -- especially in the carousel, and for this use case it's supplementing an event and I can't wrap around the SwipeableView itself without weird side-effects.

Reply:
> what do you think?

umm, I think it's fine when the code to support different platforms is minimal, which is the case here. cc @marcaaron since he could have a stronger opinion on this.

> As for adding pressable to SwipeableView in the web component..

So for the pressable [comment](https://github.com/Expensify/App/pull/9279/files#r962192842). Whats the purpose of pressable in SwipeableView for? We can't swipe on web anyway and pressing on the image zooms into it so I didn't follow why there is any change in SwipeableView.

> I wouldn't understand why there wouldn't be some drag and drop functionality to mirror the UX in mobile eventually

There could be some functionality in the future, but let's refrain from coding things that are not currently being used.

Reply:
When I have panresponder set up it takes all gestures it seems and I needed to set that up to toggle the arrows

Reply:
bump @marcaaron

Reply:
I think yes it's right that we shouldn't have platform specific code here. But we don't necessary have to rewrite the component. This is the same problem that the `KeyboardShortcut` lib solves I think. On native, we have a noop and web we add event listeners.

https://github.com/Expensify/App/blob/56dcf61a18dc391b6f52f05d05cb975f8352b63a/src/libs/KeyboardShortcut/index.native.js#L5-L12